profiles/permissions.py

```diff
@@ -26,10 +26,18 @@ def has_permission(self, request, view):
         return request.user == view._profile.person
 
 
+class OnlyRequestUserOrAdmin(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        if request.method in SAFE_METHODS and user.is_staff:
+            return True
+        return str(request.user.id) == str(request.data.get("user"))
+
+
 class IsOwnCompany(BasePermission):
     def has_permission(self, request, view):
         user = request.user
-        pk = request.data.get("company_pk")
+        pk = request.data.get("company")
```

Comment:
The same issue as above.

I'd recommend to:
1. Put company ID in URL.
2. Extract company in a cached property (`@cached_property` decorator) in a view. If it's detailed view, `get_object()` can be overridden to use value from this property too.
3. In permission class, use this property `view._company`.

Thus, permission class will not have to deal with `request.data`.

Reply:
url /saved-list - for create-list functionality. 
url /saved-list/<pk> for delete functionality.
To put id company (not saved-company object id!) to the url with create functionality doesn't seem ok for rest